msal/managed_identity.py

```diff
@@ -316,6 +316,17 @@ def _scope_to_resource(scope):  # This is an experimental reasonable-effort appr
     return scope  # There is no much else we can do here
 
 
+def _get_arc_endpoint():
+    if "IDENTITY_ENDPOINT" in os.environ and "IMDS_ENDPOINT" in os.environ:
+        return os.environ["IDENTITY_ENDPOINT"]
```

Comment:
If detecting `/var/opt/azcmagent/bin/himds` and `%ProgramFiles%\AzureConnectedMachineAgent\himds.exe` is a reliable method, why bother checking `IDENTITY_ENDPOINT` and `IMDS_ENDPOINT`?

Reply:
Indeed, that is an open topic. @neha-bhargava asked the Arc team specifically on this but received no response.

The current implementation here aligns with the implementation in MSAL .Net. In any case, this subtlety shall be considered as an implementation detail, which might be adjusted in the future again.

Reply:
Agree with @rayluo. Added this in addition to env variable detection since env variables detection have been a success in case of windows. Might remove the env variable detection in future.